File: 设计模式ProgrammingMode/Bridge/bridge/src/crayon.java
public class crayon {
    public static void main(String[] args){
        Crayon redBig = new RedBigCrayon();
        Crayon blueSmall = new BlueSmallCrayon();
        redBig.paint();
        blueSmall.paint();
    }
}

class Crayon{

    protected String size;
    protected String color;
    public void paint() {
        System.out.println(size + color + " is painting");
    }
};

class RedBigCrayon extends Crayon{
    public void setConfig(){
        size = "big";
        color = "red";
        }
    public RedBigCrayon(){
        setConfig();
        }
    };

class BlueSmallCrayon extends Crayon{
    public void setConfig(){
        size = "small";
        color = "blue";
        }
    public BlueSmallCrayon(){
        setConfig();
        }
    };

File: 设计模式ProgrammingMode/Bridge/crayon.cpp
//
// Created by Planck Chang on 2022/10/9.
//

#include <iostream>
#include <string>
using namespace std;

class Crayon{
protected:
    string size;
    string color;
public:
    void paint(){
        cout << size << " " << color << " is painting\n";
    }
    virtual void setConfig() = 0;
};

class RedBigCrayon:public Crayon{
public:
    void setConfig() override{
        size = "big";
        color = "red";
    }
    RedBigCrayon(){
        setConfig();
    }
};

class BlueSmallCrayon:public Crayon{
public:
    void setConfig(){
        size = "small";
        color = "blue";
    }
    BlueSmallCrayon(){
        setConfig();
    }
};

int main(){
    Crayon * smallblue = new BlueSmallCrayon();
    smallblue->paint();
    Crayon * bigred = new RedBigCrayon();
    bigred->paint();
}